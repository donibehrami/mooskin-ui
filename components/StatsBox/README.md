# MooSkin UI - StatsBox Component

___

### Usage

To start using the StatsBox Component first you have to Import it

```
Import {StatsBox, StatsNumber, StatsResult, StatsTitle} from '@moosend/mooskin';
```
or modular import
```
// JS
import StatsBox, {StatsNumber, StatsResult, StatsTitle} from '@moosend/mooskin/lib/StatsBox';

// CSS
import '@moosend/mooskin/lib/StatsBox/style.css';
```

And then you can simply start using it by typing

```
<StatsBox>
    <StatsTitle>Title</StatsTitle>
    <StatsNumber>Some percentage or number</StatsNumber>
    <StatsResult>whatever else</StatsResult>
</StatsBox>

```

<div class="playground-doc">

## Supported attributes for StatsBox, StatsTitle, StatsNumber and StatsResult

* `className` - main element css class
* `style` - main element style

</div>

Allthough these attributes are supported, all of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)