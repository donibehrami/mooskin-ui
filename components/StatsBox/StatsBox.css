.statsBox{
    display: flex;
    border-radius: 8px;
    flex-grow: 1;
}

.statsTitle{
    padding: 15px 10px;
    border-bottom-right-radius: 8px;
    border-top-right-radius: 8px;
    transform: rotate(-180deg);
    writing-mode: vertical-lr;
    font-size: 14px;
    justify-content: center;
    align-items: center;
    display: flex;
}

.statsContent {
    /* flex-shrink: 1;
    flex-grow: 1; */
    flex-grow: 1;
    padding: 20px;
    background: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
}
