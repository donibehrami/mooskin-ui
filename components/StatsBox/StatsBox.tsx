import * as React from 'react';

import styles from './StatsBox.css';

export interface IStatsBoxProps {
    /** css class */
    className?: string;

    /** stast box side title */
    sideTitle: string;

    /** stast box side color */
    sideColor: string;

    /** stast box side text color */
    sideTextColor: string;

    /** adds href to side title */
    sideTitleHref?: string;

    /** main element style */
    style?: React.CSSProperties;
}

export default class StatsBox extends React.Component<IStatsBoxProps, {}>{

    static defaultProps: Partial<IStatsBoxProps> = {
        className: '',
        sideTextColor: '#fff',
        style: {}
    };

    static displayName = 'StatsBox';

    render(){

        const {className, sideColor, sideTextColor, sideTitle, style, sideTitleHref} = this.props;

        const sideTitleComponent = (
            <div className={styles.statsTitle} style={{background: sideColor, color: sideTextColor}}>
                {sideTitle}
            </div>
        );

        const sideTitleAnchor = sideTitleHref ? (
            <a href={sideTitleHref} className={styles.titleAnchor} target="_blank">
                {sideTitleComponent}
            </a>
        ) : sideTitleComponent;

        return (
            <div style={style} className={`${styles.statsBox} ${className}`}>
                {sideTitleAnchor}
                <div className={styles.statsContent}>
                    {this.props.children}
                </div>
            </div>
        );
    }

}
