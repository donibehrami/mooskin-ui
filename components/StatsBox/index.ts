import StatsBox, {StatsNumber, StatsResult, StatsTitle} from './StatsBox';

StatsBox.StatsNumber = StatsNumber;
StatsBox.StatsResult = StatsResult;
StatsBox.StatsTitle = StatsTitle;

export default StatsBox;
export {StatsNumber};
export {StatsResult};
export {StatsTitle};
