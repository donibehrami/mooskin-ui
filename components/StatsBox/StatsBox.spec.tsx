import * as React from 'react';
import StatsBox, {StatsNumber, StatsResult, StatsTitle} from './StatsBox';

import { shallow } from 'enzyme';

describe('StatsBox', () => {

    test('renders correctly', () => {

        const tree = shallow(
            <StatsBox>
                <StatsTitle>Unique Opens</StatsTitle>
                <StatsNumber>100%</StatsNumber>
                <StatsResult>5 total</StatsResult>
            </StatsBox>
        );
        expect(tree).toMatchSnapshot();
    });
});
