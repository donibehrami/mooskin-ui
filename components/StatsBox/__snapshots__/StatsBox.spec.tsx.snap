// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsBox StatsNumber renders correctly 1`] = `
<div
  className="statsNumber "
  style={Object {}}
>
  100%
</div>
`;

exports[`StatsBox StatsResult renders correctly 1`] = `
<div
  className="statsResult "
  style={Object {}}
>
  5 total
</div>
`;

exports[`StatsBox StatsTitle renders correctly 1`] = `
<div
  className="statsTitle "
  style={Object {}}
>
  Unique Opens
</div>
`;

exports[`StatsBox renders correctly 1`] = `
<div
  className="statsBox "
  style={Object {}}
>
  <StatsTitle
    className=""
    style={Object {}}
  >
    Unique Opens
  </StatsTitle>
  <StatsNumber
    className=""
    style={Object {}}
  >
    100%
  </StatsNumber>
  <StatsResult
    className=""
    style={Object {}}
  >
    5 total
  </StatsResult>
</div>
`;
